server/router/src/main/java/org/infinispan/server/router/router/impl/hotrod/handlers/util/SslUtils.java

```diff
@@ -0,0 +1,26 @@
+package org.infinispan.server.router.router.impl.hotrod.handlers.util;
+
+import io.netty.handler.ssl.ClientAuth;
+import io.netty.handler.ssl.IdentityCipherSuiteFilter;
+import io.netty.handler.ssl.JdkSslContext;
+import io.netty.handler.ssl.SslContext;
+import org.infinispan.commons.util.SslContextFactory;
+
+import javax.net.ssl.SSLContext;
+import java.security.NoSuchAlgorithmException;
+import java.util.Arrays;
+import java.util.Optional;
+
+public enum SslUtils {
+    INSTANCE;
```

Comment:
I've been using Singleton pattern from Josh Bloch's Effective Java: http://www.informit.com/articles/article.aspx?p=1216151&seqNum=3

I guess the arguments mentioned in the book are still valid (even though the book is quite old). Are there any arguments why I shouldn't use enum here?